paddle/fluid/operators/cinn_launch_op.cc

```diff
@@ -62,90 +62,105 @@ void DebugCinnCompiledResult(const CinnCompiledObject& result) {
-                 });
-  return result;
+bool CinnLaunchContext::IsVariableUsed(const std::string& paddle_name) {
+  return paddle2cinn_varmap_.count(paddle_name) > 0 &&
+         cinn_variable_names_.count(paddle2cinn_varmap_.at(paddle_name)) > 0;
+}
+
+CinnTensor CinnLaunchContext::GetCinnTensor(const std::string& var_name) {
+  PADDLE_ENFORCE_GT(cinn_variable_names_.count(var_name), 0,
+                    platform::errors::NotFound(
+                        "Variable(%s) not found in cinn scope.", var_name));
+  return cinn_scope_->GetTensor(var_name);
+}
+
+std::vector<std::string> CinnLaunchContext::GetInternalVariableNames() {
+  std::unordered_set<std::string> all_parameters(cinn_variable_names_);
+  std::for_each(paddle2cinn_varmap_.begin(), paddle2cinn_varmap_.end(),
+                [&all_parameters, this](const auto& name_pd2cinn) {
+                  if (IsVariableUsed(name_pd2cinn.first)) {
+                    all_parameters.erase(name_pd2cinn.second);
+                  }
+                });
+  return {all_parameters.begin(), all_parameters.end()};
+}
+
+void CinnLaunchContext::MutableTensorData(const std::string& var_name,
+                                          const platform::Place& place,
+                                          LoDTensor* paddle_tensor,
+                                          bool is_internal_var) {
+  auto cinn_name = var_name;
+  if (!is_internal_var) {
+    PADDLE_ENFORCE_EQ(IsVariableUsed(var_name), true,
+                      platform::errors::InvalidArgument(
+                          "Paddle variable(%s) not used by cinn", var_name));
+    cinn_name = paddle2cinn_varmap_.at(var_name);
+  }
+
+  auto cinn_tensor = GetCinnTensor(cinn_name);
+  // TODO(CtfGo): support mutable corresponding c++ type after CINN ready
+  paddle_tensor->mutable_data<float>(
+      framework::make_ddim(cinn_tensor->shape().data()), place);
 }
 
-void CheckTensorEquivalent(const std::string& paddle_name,
-                           const LoDTensor* paddle_tensor,
-                           const CinnTensor& cinn_tensor) {
+void CinnLaunchContext::CheckTensorEquivalent(const std::string& paddle_name,
+                                              const LoDTensor* paddle_tensor,
+                                              const CinnTensor& cinn_tensor) {
   PADDLE_ENFORCE_EQ(
       paddle_tensor->IsInitialized(), true,
       platform::errors::InvalidArgument(
-          "The tensor in variable(%s) is not initialized.", paddle_name));
+          "Tensor in variable(%s) is not initialized.", paddle_name));
 
   // check dimension
   auto cinn_dims = framework::make_ddim(cinn_tensor->shape().data());
   PADDLE_ENFORCE_EQ(paddle_tensor->dims(), cinn_dims,
                     platform::errors::InvalidArgument(
-                        "The tensor dimension in variable(%s) "
-                        "is not equivalent, paddle is [%s] "
+                        "Tensors dimension in variable(%s) "
```

Comment:
NIT (Not important): I saw you noticed the English grammar error, but the change is not correct ...

`Tensors' dimensions` or `Dimensions of tensors`. The word "Dimensions" is the noun here and the verb is "are"

Reply:
fixed